protocol/communities/manager.go

```
@@ -3220,43 +3156,31 @@ func (m *Manager) BanUserFromCommunity(request *requests.BanUserFromCommunity) (
 	return community, nil
 }
 
-// Apply events to raw community
-func (m *Manager) initializeCommunity(community *Community) error {
-	err := community.updateCommunityDescriptionByEvents()
-	if err != nil {
-		return err
-	}
-
-	if m.transport != nil && m.transport.WakuVersion() == 2 {
-		topic := community.PubsubTopic()
-		privKey, err := m.transport.RetrievePubsubTopicKey(topic)
+func (m *Manager) dbRecordBundleToCommunity(r *CommunityRecordBundle) (*Community, error) {
+	return recordBundleToCommunity(r, &m.identity.PublicKey, m.installationID, m.logger, m.timesource, func(community *Community) error {
+		err := community.updateCommunityDescriptionByEvents()
 		if err != nil {
 			return err
 		}
-		community.config.PubsubTopicPrivateKey = privKey
-	}
 
-	// Workaround for https://github.com/status-im/status-desktop/issues/12188
-	HydrateChannelsMembers(community.IDString(), community.config.CommunityDescription)
+		if m.transport != nil && m.transport.WakuVersion() == 2 {
+			topic := community.PubsubTopic()
+			privKey, err := m.transport.RetrievePubsubTopicKey(topic)
+			if err != nil {
+				return err
+			}
+			community.config.PubsubTopicPrivateKey = privKey
+		}
```

maybe return an error if waku version is different?

---

Wanted to preserve the same behavior as was before. Once we get rid off wakuv1 we should make it an error I believe.